### PR TITLE
rgw: correct multipart meta location when upload to non-standard storage class

### PR DESCRIPTION
**Pull request by dongbula**
Fixes: https://tracker.ceph.com/issues/44511
Signed-off-by: lvshuhua <lvshuhua@cmss.chinamobile.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dongbula** [comment]
@cbodley hi, could you help to review this , thank you

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -3935,6 +3935,7 @@ void RGWPutObj::execute()
       return;
     }
     pdest_placement = &upload_info.dest_placement;
+    s->bucket_info.placement_rule = upload_info.dest_placement;
```
Comment (cbodley) [inline src/rgw/rgw_op.cc]: we shouldn't mutate the bucket_info for this. can you add a new `meta_placement_rule` argument to the `MultipartObjectProcessor` constructor for `raw_meta_obj`?

Comment (dongbula) [reply src/rgw/rgw_op.cc]: `RGWRados::get_obj_head_ref` that called by `RGWRados::Object::write::_do_write_meta` accept a `RGWBucketInfo ` type parameter to locate meta obj，if we add a `meta_placement_rule` for `MultipartObjectProcessor `,  may also need to construct a new bucket_info before calling `RGWRados::get_obj_head_ref`, or change those functions' interface form on call stack.
we can use some structs to pass dest_placement info, such as obj_op->target.bucket_info or obj_op.meta->manifest.head_placement_rule(in MultipartObjectProcessor::complete), what you about it?

Comment (dongbula) [reply src/rgw/rgw_op.cc]: updated a version

Comment (dongbula) [reply src/rgw/rgw_op.cc]: @cbodley hi, can we continue this pr?

**Pull request closed without merging**